diff --git a/Modules/socketmodule.c b/Modules/socketmodule.c
index 4ddba8b..f976876 100644
--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -5991,7 +5991,7 @@ Return the IP address (a string of the form '255.255.255.255') for a host.");
 #endif
 
 
-#if defined(HAVE_GETHOSTBYNAME_R) || defined (HAVE_GETHOSTBYNAME) || defined (HAVE_GETHOSTBYADDR)
+#if defined(HAVE_GETHOSTBYNAME_R) || defined (HAVE_GETHOSTBYNAME) || defined (HAVE_GETHOSTBYADDR) || defined (HAVE_GETNAMEINFO)
 static PyObject*
 sock_decode_hostname(const char *name)
 {
@@ -6004,7 +6004,9 @@ sock_decode_hostname(const char *name)
     return PyUnicode_FromString(name);
 #endif
 }
+#endif
 
+#if defined(HAVE_GETHOSTBYNAME_R) || defined (HAVE_GETHOSTBYNAME) || defined (HAVE_GETHOSTBYADDR)
 /* Convenience function common to gethostbyname_ex and gethostbyaddr */
 
 static PyObject *
